### DISCUSSION TITLE
Model relationship across multiple entities?

### DISCUSSION THREAD

**Discussion by itsame-luigi** [mikro-orm/mikro-orm#2526] [Q&A]
Lets say you have the following four entities:

```js
@Entity()
class Foo {
  @PrimaryKey()
  id: string;

  @OneToMany(() => FooBar, obj => obj.foo)
  foobars = new Collection<FooBar>(this);

  // (a)
  // @OneToMany(...)
  // bazes = new Collection<Baz>(this);
}

@Entity()
class Bar {
  @PrimaryKey()
  id: string;

  @OneToMany(() => FooBar, obj => obj.bar)
  foobars = new Collection<FooBar>(this);

  // (b)
  // @OneToMany(...)
  // bazes = new Collection<Baz>(this);
}

@Entity() 
class FooBar {
  @ManyToOne(() => Foo, { eager: true, primary: true })
  foo: Foo;

  @ManyToOne(() => Bar, { eager: true, primary: true })
  bar: Bar;

  @OneToMany(() => Baz, obj => obj.foobar)
  bazes = new Collection<Baz>(this);
}

@Entity()
class Baz {
  @PrimaryKey()
  id: string;

  @ManyToOne(() => FooBar, { eager: true })
  foobar: FooBar;
}
```

Is it possible to use `OneToMany` to model the relationship at (a) for `Foo` -> `FooBar` -> `Baz` (or at (b) for `Bar` -> `FooBar` -> `Baz`)?

**Comment by B4nan** [comment]
Nope, you need to work thru the foobar entity. I was thinking about allowing this (it would be m:n, not 1:m btw), but it would be only for reading (the collection would be readonly).

**Reply by itsame-luigi** [reply]
It doesn't seem to be many to many. A `Baz` can only have one `foobar`, which can only have one `foo` and one `bar`, so a `Baz` can only have one `foo` and a `Foo` can have many `Baz`.

**Reply by itsame-luigi** [reply]
But I agree, it would probably be readonly
